File: BSF-Backend/src/main/java/com/bsf/gub/model/CommitteeMember.java
package com.bsf.gub.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "committee_members")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CommitteeMember {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String fullName;
    
    @Column(nullable = false)
    private String position; // President, Vice President, General Secretary, etc.
    
    @Column(nullable = false)
    private String email;
    
    private String phone;
    
    private String studentId;
    
    private String department;
    
    private String imageUrl; // Profile photo URL
    
    private String linkedinUrl;
    
    private String facebookUrl;
    
    @Column(columnDefinition = "TEXT")
    private String bio;
    
    @Column(nullable = false)
    private String year; // 2024, 2025, etc. - for yearly tracking
    
    @Column(nullable = false)
    private Integer displayOrder = 0; // For custom ordering
    
    @Column(nullable = false)
    private Boolean active = true;
    
    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: BSF-Backend-Node/src/models/CommitteeMember.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const CommitteeMember = sequelize.define('CommitteeMember', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  fullName: {
    type: DataTypes.STRING(255),
    allowNull: false,
    field: 'full_name'
  },
  position: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  email: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  phone: {
    type: DataTypes.STRING(20),
    allowNull: true
  },
  studentId: {
    type: DataTypes.STRING(50),
    allowNull: true,
    field: 'student_id'
  },
  department: {
    type: DataTypes.STRING(100),
    allowNull: true
  },
  imageUrl: {
    type: DataTypes.STRING(500),
    allowNull: true,
    field: 'image_url'
  },
  linkedinUrl: {
    type: DataTypes.STRING(500),
    allowNull: true,
    field: 'linkedin_url'
  },
  facebookUrl: {
    type: DataTypes.STRING(500),
    allowNull: true,
    field: 'facebook_url'
  },
  bio: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  year: {
    type: DataTypes.STRING(10),
    allowNull: false
  },
  displayOrder: {
    type: DataTypes.INTEGER,
    allowNull: false,
    defaultValue: 0,
    field: 'display_order'
  },
  active: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: true
  }
}, {
  tableName: 'committee_members',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at',
  indexes: [
    {
      fields: ['year']
    },
    {
      fields: ['active']
    },
    {
      fields: ['display_order']
    },
    {
      fields: ['year', 'active', 'display_order']
    }
  ]
});

module.exports = CommitteeMember;
